src/GafferAppleseed/IECoreAppleseedPreview/Renderer.cpp

```diff
@@ -1299,6 +1299,21 @@ class AppleseedPrimitive : public AppleseedEntity
 			m_objectInstance->get_parameters().insert( "visibility", appleseedAttributes->m_visibilityDictionary );
 
 			// todo: support edits of smooth normals and tangents attribute.
+
+			if( isInteractiveRender() )
+			{
+				// We need to re-create object instances after edits.
+				asr::ObjectInstance *objI = m_objectAssembly->object_instances().get_by_name( m_objectInstance->get_name() );
```

Comment:
Dumb question : how is `object_instances().get_by_name( m_objectInstance->get_name() )` different from `m_objectInstance`?

Reply:
I'll check. We use this pattern when inserting things into containers because the containers take ownership of the pointers, but it is very likely that here it is not needed.